bindings/matrix-sdk-crypto-ffi/src/machine.rs

```diff
@@ -66,6 +72,34 @@ pub struct KeyRequestPair {
     pub key_request: Request,
 }
 
+/// The result of a signature check of a signed JSON object.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct SignatureCheckResult {
```

Comment:
If I understand this improvement correctly, clients want to have more info about why backup is trusted / not trusted, but they also still want the overal computed `trusted` state as a simple boolean, otherwise they have to do the comparison of 3 enum states themselves

Reply:
Hmm that would require us to do one of the following:
- change the SignatureCheckResult from being a `dictionary` to an `interface` in the UDL file
- add a boolean `trusted` field to the struct which calls the rust method that does the comparison

Any preference?

Reply:
I think the second should suffice, i.e. `trusted` field on the struct which is computed when we create / return `SignatureVerification`

Reply:
Alrighty, done in https://github.com/matrix-org/matrix-rust-sdk/pull/1217/commits/2fcfd2226944fabf7c040ef9a4f794fd1c15398f.